### ISSUE TITLE
Data

### ISSUE THREAD

**Issue by hkmztrk** [Shen-Lab/DeepAffinity#1]
Hello,

I was looking for channel, GPCR, ER and kinase datasets with train/test splits, but couldn't find them in  In "baseline" folder. Could you please help me?

**Comment by Shen-Lab** [comment]
Thank you for your interest in our data.  These four classes of proteins are completely withheld from the original BindingDB data as generalization sets and the rest are split into training (70%) and test (30%) sets.  In other words, the training set does not contain any measurement for ion channels, GPCR, ER, or receptor tyrosine kinase; and models trained using the training set are assessed on both the test set and the four generalization sets.  A revision has been made in README under data to avoid confusion.

**Comment by hkmztrk** [comment]
Thank you for your help!

**Comment by Shen-Lab** [comment]
Thanks again for your interest.  Please feel free to let us know whenever you have a question about our data or method.

**Comment by hkmztrk** [comment]
Hello,

I have another question regarding to  training and test processes. As far as I understand, for instance for IC50 dataset, you keep the training data same, but change test set such as "testing,  ER, IC, GPCR" etc., right? 

How many times an experiment is repeated? For instance, training on the training set and testing on GPCR set, your results in the tables are given for a single run or are given for an average of multiple runs?

Thank you!

**Comment by mostafakarimi71** [comment]
Thank you for your interest in our training process. We implemented an ensemble of K deep learning models (with K=3*10) for each of our developed model (joint/separate + 3 different attention mechanism and ....). Final prediction was the average of all these models' predictions (in non-weighted way). Finally, RMSE/MSE is calculated between the final prediction and the true labels. However due to computational limitation, we didn't repeat this process for several times.

Each of K ensemble model was trained once on the train data and tested once on the "testing, ER,.." data.

**Comment by hkmztrk** [comment]
Thanks for your reply.

Does 10 in (K = 3* 10) refer  here? 
"averaging a “parameter ensemble” of 10 models derived in the last 10 epochs"?  Could you please elaborate about parameter ensemble and why the last 10 epochs were chosen? I am also confused about 3 in K = 3 *10.

**Comment by mostafakarimi71** [comment]
With the same spirit from random forest that ensemble methods can reduce the variance, we consider an ensemble of deep learning models. We compare two types of ensemble methods:

1) Within a model: Use N trained model from different epochs to reduce the variance. We used the last 10 epochs which was chosen empirically. This ensemble almost comes free and you only needs to save those models
2) Across models:  Use N different models (different hyper-parameters) to reduce the variance. We change the number of neurons in fully connected layers (There are other hyper-parameters that can be changed as well). But, you need to actually train N different model. However, across models ensemble can reduce the variance drastically. 

At last we combine both of these ideas and get 3(across models)* 10(within model) ensemble models.

**Comment by hkmztrk** [comment]
Thank you @mostafakarimi71 !
